update promlog to latest version

alexander-yu opened the pull request:
Revendor `prometheus/common` for the latest promlog version, in order to support changes from prometheus/common#136 (and prometheus/common#156); this resolves #3219 (i.e. adding JSON support for log output format).

**Note:** when revendoring (via `go mod`) promlog, this ended up retrieving the latest version of the entire `prometheus/common` repo; is there a good way of having a more granular update? I'm unsure what effects there are for updating the other parts of the `prometheus/common` repo.

simonpasquier commented:
> when revendoring (via go mod) promlog, this ended up retrieving the latest version of the entire prometheus/common repo;

This is working as intended regarding Go modules. I would say that in general, the old way of mixing different revisions for packages from the same repository (module) was more prone to errors.

brian-brazil commented on cmd/prometheus/main.go:
```diff
@@ -101,11 +101,15 @@ func main() {
 			Registerer: prometheus.DefaultRegisterer,
 		},
+		promlogConfig: promlog.Config{
+			Level:  new(promlog.AllowedLevel),
```
We shouldn't have to list these here, as that goes against the goal of not having to update code when there's a new setting.

alexander-yu replied:
Oh right; I just made a PR for promlog's `AddFlags` method that also initializes these for the config, which avoids this issue.

See prometheus/common#159.

brian-brazil commented on documentation/examples/remote_storage/remote_storage_adapter/main.go:
```diff
@@ -104,7 +105,16 @@ func main() {
+	}
+
+	logger := promlog.New(&promlog.Config{
+		Level:  &logLevel,
```
Ditto

alexander-yu commented:
Updated again so we only need to initialize the config; also modified `remote_storage_adapter/main.go` to use kingpin, since promlog's `AddFlags` method requires a kingpin application.

brian-brazil commented:
That looks good bar Simon's comments.

alexander-yu commented on documentation/examples/remote_storage/remote_storage_adapter/main.go:
```diff
@@ -15,31 +15,35 @@
 	"github.com/prometheus/prometheus/documentation/examples/remote_storage/remote_storage_adapter/opentsdb"
 	"github.com/prometheus/prometheus/prompb"
+
+	"gopkg.in/alecthomas/kingpin.v2"
```
Done

alexander-yu commented on documentation/examples/remote_storage/remote_storage_adapter/main.go:
```diff
@@ -15,31 +15,35 @@
 	influx "github.com/influxdata/influxdb/client/v2"
 
 	"github.com/prometheus/common/promlog"
+	"github.com/prometheus/common/promlog/flag"
```
Done

alexander-yu commented:
Cleaned up the imports as per Simon's CR.

simonpasquier commented:
LGTM
